File: binary_search.py
def binary_search(array,item):
	left = 0
	right = len(array)-1
	while(left<right):
		mid = (left+right)//2
		if(item==array[mid]):
			return (True,mid)
		if(item<array[mid]):
			right = mid-1
		else:
			left = mid+1
	return (False,-1)


arr = [3,4,6,10,24,26,34]
print(binary_search(arr,55))

File: binary_search.cpp
#include <iostream>
using namespace std;


int my_binary_search(int *array, int element, int array_size)
{
	int left = 0;
	int right = array_size-1;
	while(left<right){
		int mid = (left+right)/2;
		if(element==array[mid]){
			return mid;
		}else if(element<array[mid]){
			right = mid-1;
		}else{
			left = mid+1;
		}
	}
	return -1;
}

int main(){
	int array[] = {1,4,3,5,6,10,21};
	int result = my_binary_search(array,10,7);
	cout << result << endl;
}
